documents/extension/funding/funding.go

```diff
@@ -31,3 +31,17 @@ func (f *Data) initFundingFromData(data *clientfundingpb.FundingData) {
 
 	}
 }
+
+func (f *Data) getClientData() *clientfundingpb.FundingData {
+	clientData := new(clientfundingpb.FundingData)
+
+	types := reflect.TypeOf(*f)
+	values := reflect.ValueOf(*f)
+	for i := 0; i < types.NumField(); i++ {
+		n := types.Field(i).Name
+		v := values.FieldByName(n).Interface().(string)
+		reflect.ValueOf(clientData).Elem().FieldByName(n).SetString(v)
```

Comment:
I am not sure how safe is to assume that all our API fields are strings. They could be wrapped as well as complex structures of strings. Let's make sure we add a comment about the limitations of this converter.